### PR TITLE
EBP-383: Fix runtime cgo panic by changing CCSMP opaque pointer typdefs

### PR DESCRIPTION
**Pull request by TrentDaniel**
Added compile directive to solClient.h so that when the Go API links against CCSMP, it treats all opaque pointers as uintptrs (effectively longs) instead of as void pointers, except for solClient_opaquePtr_pt. I couldn't figure out how to get around solClient_opaquePtr_pt because certain CCSMP functions like solClient_msg_getBinaryAttachmentPtr require a pointer to an opaque pointer. This was all done to avoid the cgo runtime from panicking when it does a runtime check of a C/Go boundary and believes that the 'void *' type that the opaque pointers used to be typedef'd to was an unpinned pointer. As mentioned, the change to the typdef of opaque pointers in solClient.h will prevent the cgo runtime from panicking since it will evaluate the opaque pointers as longs/ints instead of unpinned references. It's unclear whether or not the solClient_opaquePtr_pt will become a problem in the future, but it is actually pointing to data whereas the other solClient_opaque*_pt types are not, which the cgo runtime should be able to discern. The other changes made in this commit were in response to the typdef change, and isolating more memory allocation of configuration structs like the sessionCreateFuncInfo to C in order to mitigate the risk of bad references being accessed by CCSMP.

### CODE REVIEW

File: internal/impl/publisher/request_reply_message_publisher_impl.go [hunk 1]
```diff
@@ -877,12 +877,12 @@ func (publisher *requestReplyMessagePublisherImpl) handleReplyMessage(msgP core.
...
 	corEntry, ok := publisher.requestCorrelationMap[correlationID]
 	if !ok {
-		publisher.logger.Debug(fmt.Sprintf("Received reply message[%p] with correlationID[%s] without correlation entry for publisher", msgP, correlationID))
+		publisher.logger.Debug(fmt.Sprintf("Received reply message[0x%x] with correlationID[%s] without correlation entry for publisher", msgP, correlationID))
 		return false
 	}
 	if corEntry.received {
 		// return false to return the message
-		publisher.logger.Debug(fmt.Sprintf("Received reply message[%p] with correlationID[%s] that already has response", msgP, correlationID))
+		publisher.logger.Debug(fmt.Sprintf("Received reply message[0x%x] with correlationID[%s] that already has response", msgP, correlationID))
```
Comment (oodigie) [inline internal/impl/publisher/request_reply_message_publisher_impl.go:885]: @TrentDaniel, please why did you change the Sprintf formatting verb from `%p` to `%x` ?

**Review by oodigie — COMMENTED** [review]
I left a comment.

**Comment by oodigie** [comment]
@TrentDaniel, also I see a lot of failures on Jenkins - https://jenkins2.solacedev.net/job/solacedev.pubsubplus-go-client/job/EBP-383/3/#showFailuresLink

**Review by RagnarPaulson — CHANGES_REQUESTED** [review]

File: internal/impl/publisher/request_reply_message_publisher_impl.go (hunk 1 above)
Comment (TrentDaniel) [reply internal/impl/publisher/request_reply_message_publisher_impl.go:885]: @oodigie , Ragnar is correct, the formatting verb no longer matched the type. I could've instead done some conversions to get a pointer just for the print statement, but that would have require multiple casts and new objects to be created. Since it was only two lines that needed to be printed, I just formatted the integer in hexadecimal, and prefixed it with a hardcoded string of `0x` so that it would appear as before. The prefix is required because while the `%p` verb will automatically prepend that prefix to the hex repr of the int, the `%x` will not. This can be observed through this example:  https://go.dev/play/p/7v9BwWInXIn.

File: internal/ccsmp/lib/include/solclient/solClient.h [hunk 2]
```diff
@@ -18,6 +18,7 @@
 #include <limits.h>
 #include <stdio.h>
 #include <wchar.h>
+#include <stdint.h>
```
[the remark this answers is not in the document]
Comment (TrentDaniel) [reply internal/ccsmp/lib/include/solclient/solClient.h:21]: @RagnarPaulson Do we make a commitment to customers that our header files will match those that we used to build the binaries that are packaged with the API? If that's the case then we might have to get a new CCSMP build before we can release this. :/

[the remark this answers is not in the document]
Comment (RagnarPaulson) [reply internal/ccsmp/lib/include/solclient/solClient.h:21]: It's good practice, but we don't make that commitment.   Also I was never planning to create a C API variant for Go on all the go platforms.  So no matter what, the C API will continue to be built with 'void *' and it's only the Go Wrapper API that treats it as a uintptr_t.  

From the testing this week we have empirical evidence that this difference doesn't cause a problem.

[the remark this answers is not in the document]
Comment (TrentDaniel) [reply internal/ccsmp/lib/include/solclient/solClient.h]: Just pushed the fix

**Review by RagnarPaulson — APPROVED** [review]
All looks great now. Thanks.

**Review by sdewilde — APPROVED** [review]
No comment on the change itself but the changes have run well in QA tests.  Run our platform test on alpine with no issues and the test failures we saw in the feature test were infra related and passed on rerun with other resources.  Looks good!

**Comment by oodigie** [comment]
@TrentDaniel, I still see a lot more failures on Jenkins - https://jenkins2.solacedev.net/job/solacedev.pubsubplus-go-client/job/EBP-383/4/#showFailuresLink .

**Comment by TrentDaniel** [comment]
> @TrentDaniel, I still see a lot more failures on Jenkins - https://jenkins2.solacedev.net/job/solacedev.pubsubplus-go-client/job/EBP-383/4/#showFailuresLink . Please can you at least verify that the test failures are not related to the changes in this PR ?

I've verified that the test failures are not caused by the changes in this PR. Summary:
- 4 are some kind of SEMP timeout, probably to do with bad broker setup and definitely unrelated to API changes
- 1 is a race condition, with ticket EBP-401
- 1 is known test failure that you are currently working on
- The rest are known test failures from a bottleneck in one of our test nodes
